### PR TITLE
feat: Add a Color Selector Control #30

### PR DESCRIPTION
**Pull request by MichaelThomas0721**
## Description

This PR adds a color selector control. This control allows users to use the color input type to select a color as a string (hex format). This grants users the ability to quickly choose/change colors on the fly without needing to have pre selected them. Personally I was looking for a way to quickly test out new colors and previously I was using a separate website's color selector to pick a color and then copy pasting the hex code into a string control. This makes that process a lot more streamlined and reactive.
#573 #30 

### Additional context

I am new to VUE and was struggling with the best way to update the state, when I was just using the emit function that I saw in other controls, it was super slow, probably because it was doing it so much. I ended up implemented a throttle function so it will only shoot every 15ms which fixed it. If there is a better way then let me know. I tried to follow the guidelines as best as possible, let me know if anything needs to be changed :)

### What is the purpose of this pull request? 

- [ ] Bug fix
- [X] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [X] If it's a new feature, provide a convincing reason to add it. *Ideally, you should open a suggestion issue first and have it approved before working on it.*
- [X] Read the [Contributing Guidelines](https://github.com/Akryum/histoire/blob/main/CONTRIBUTING.md).
- [X] Read the [Pull Request Guidelines](https://github.com/Akryum/histoire/blob/main/CONTRIBUTING.md#pull-request-guidelines) and follow the [Commit Convention](https://github.com/Akryum/histoire/blob/main/.github/commit-convention.md).
- [X] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [X] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

**Comment by MichaelThomas0721** [comment]
I've never used cypress before so I could be wrong but it seems that since the color selector control also uses a text input it is returning two when looking for the HstText, I will fix this shortly and add a test case for the color selector control as well.

**Comment by MichaelThomas0721** [comment]
I had to change the way that cypress gets the text control since the color selector control also uses a text input. Instead of grabbing the ones that contain `input[type="text"]` (which isn't recommended under cypress best practices), it grabs it by `.contains("HstText")`. I did the same with the new color select test. This fixed the HstText test.

I also updated the amount of stories in it to 30 from 29 since I didn't account for the new Color Select Control :)

File: examples/vue3/vite.config.ts [hunk 1]
```diff
@@ -29,8 +29,11 @@ export default defineConfig({
 
   server: {
     hmr: {
-      clientPort: process.env.CODESPACES ? 443 : undefined,
```
Comment (Akryum) [inline examples/vue3/vite.config.ts:32]: Why this change?

File: packages/histoire-plugin-vue/components.d.ts [hunk 2]
```diff
@@ -155,5 +156,6 @@ declare module '@vue/runtime-core' {
     HstTokenList: typeof HstTokenList
     HstTokenGrid: typeof HstTokenGrid
     HstCopyIcon: typeof HstCopyIcon
+    HstColorSelect: typeof HstColorSelect
```
Comment (Akryum) [inline packages/histoire-plugin-vue/components.d.ts:159]: We also need to update the svelte types and code to add the new control.

**Review by Akryum — CHANGES_REQUESTED** [review]

File: examples/vue3/vite.config.ts (hunk 1 above)
Comment (MichaelThomas0721) [reply examples/vue3/vite.config.ts:32]: haha im not actually sure, I was trying to set it up on a different port so I could use it on my remote server and I guess I accidentally deleted that line, I will fix it tomorrow.

File: packages/histoire-plugin-vue/components.d.ts (hunk 2 above)
Comment (MichaelThomas0721) [reply packages/histoire-plugin-vue/components.d.ts:159]: Can do!

File: examples/vue3/vite.config.ts (hunk 1 above)
Comment (MichaelThomas0721) [reply examples/vue3/vite.config.ts:32]: I reverted the change,  I appreciate the help :)

File: packages/histoire-plugin-vue/components.d.ts (hunk 2 above)
Comment (MichaelThomas0721) [reply packages/histoire-plugin-vue/components.d.ts:159]: I added it to the svelte plugin and added a test to the svelte3 example

**Review by Akryum — APPROVED** [review]
